pkg/pulumiyaml/run.go

```diff
@@ -285,10 +289,11 @@ func (r *Runner) setDefaultProviders() error {
 }
 
 // PrepareTemplate prepares a template for converting or running
-func PrepareTemplate(t *ast.TemplateDecl, loader PackageLoader) (*Runner, syntax.Diagnostics, error) {
-	// set up context-free runner
-	r := newRunner(t, loader)
-
+func PrepareTemplate(t *ast.TemplateDecl, r *Runner, loader PackageLoader) (*Runner, syntax.Diagnostics, error) {
```

Comment:
I think there needs to be a comment explaining why you would/would not pass a runner through.